src/components/NavBar/NavigationBar.cy.tsx

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import NavigationBar from "@/components/NavBar/NavigationBar";
+import { AppRouterContext, AppRouterInstance } from "next/dist/shared/lib/app-router-context";
+// import { useRouter } from "next/navigation";
+
+describe("<NavigationBar />", () => {
+    const router: AppRouterInstance = {
+        push: () => {},
+        back: () => {},
+        forward: () => {},
+        replace: () => {},
+        prefetch: () => {},
+        refresh: () => {},
+    };
+
+    const RouterWrappedNavBar = (
+        <AppRouterContext.Provider value={router}>
+            <NavigationBar />
+        </AppRouterContext.Provider>
+    );
+
+    it("Renders on desktop", () => {
+        cy.viewport(1367, 1500);
+
+        cy.mount(RouterWrappedNavBar);
+    });
+
+    it("Renders on mobile", () => {
+        cy.viewport(360, 1000);
+        cy.mount(RouterWrappedNavBar);
+    });
+
+    it("Links exist on desktop", () => {
+        cy.viewport(1367, 1500);
+        cy.mount(RouterWrappedNavBar);
+
+        cy.get("a[href='/clients']").contains("Clients").should("exist");
+        cy.get("a[href='/lists']").contains("Lists").should("exist");
+        cy.get("a[href='/calendar']").contains("Calendar").should("exist");
+    });
+
+    it("Links exist on mobile", () => {
+        cy.viewport(360, 1000);
+        cy.mount(RouterWrappedNavBar);
+
+        cy.get("button[aria-label='Mobile Menu Button']").click();
+        cy.get("a[href='/clients']").contains("Clients").should("exist");
```

Comment:
these might test that at least one button that has `/clients`, but eg if mobile was broken the test may select the desktop (hidden) version!
is there a way of selecting the desktop div (and aliasing it) then doing e.g. `cy.get(@desktopdiv).get("a[href...`

Reply:
Have done `.filter(":visible")` in order to only select the visible ones!